client/src/components/USBController.tsx

```diff
@@ -1,53 +1,30 @@
-import React, { useState } from 'react'
-
-import useInterval from 'use-interval'
+import React, { useContext } from 'react'
+import AppContext from '../contexts/AppContext'
 
 import Button from './Button'
 import Text from './Text'
-import {
-  isPresent,
-  isMounted,
-  doMount,
-  doUnmount,
-  isAvailable,
-} from '../lib/usbstick'
+
+import { UsbDriveStatus } from '../lib/usbstick'
 
 const USBController = () => {
-  const available = isAvailable()
-  const [present, setPresent] = useState(false)
-  const [mounted, setMounted] = useState(false)
-
-  useInterval(
-    () => {
-      ;(async () => {
-        const p = await isPresent()
-        setPresent(p)
-        if (p) setMounted(await isMounted())
-      })()
-    },
-    available ? 1000 : false
-  )
+  const { usbDriveStatus: status, usbDriveEject } = useContext(AppContext)
 
-  if (!available) {
+  if (status === UsbDriveStatus.notavailable) {
     return null
   }
 
-  if (!present) {
+  if (status === UsbDriveStatus.absent) {
     return <Text>No USB</Text>
   }
 
+  if (status === UsbDriveStatus.present) {
+    return <Text>Connecting...</Text>
```

Comment:
Use proper ellipsis (`option-;` on mac) …or eventually use the `Loading` component. 😉 
```suggestion
    return <Text>Connecting…</Text>
```